File: src/main/java/club/someoneice/thursday_delight/chili/ChiliCrop.java
package club.someoneice.thursday_delight.chili;

import javax.annotation.Nullable;

import club.someoneice.thursday_delight.ItemList;
import net.minecraft.core.BlockPos;
import net.minecraft.sounds.SoundSource;
import net.minecraft.world.InteractionHand;
import net.minecraft.world.InteractionResult;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.ItemLike;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.block.CropBlock;
import net.minecraft.world.level.block.SoundType;
import net.minecraft.world.level.block.entity.BlockEntity;
import net.minecraft.world.level.block.state.BlockBehaviour;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.material.Material;
import net.minecraft.world.phys.BlockHitResult;
import vectorwing.farmersdelight.common.registry.ModSounds;

public class ChiliCrop extends CropBlock {

    public ChiliCrop() {
        super(BlockBehaviour.Properties.of(Material.PLANT).noCollission().sound(SoundType.CROP));
    }

    @Override
    protected ItemLike getBaseSeedId() {
        return ItemList.Chili.get();
    }

    @Override
    public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
        if (this.isMaxAge(state)) {
            // player.getInventory().add(new ItemStack(ItemList.Chili.get(), new Random().nextInt(1, 3)));
			popResource(level, pos, new ItemStack(ItemList.Chili.get(), level.random.nextInt(1, 3)));
            level.playSound(null, pos, ModSounds.ITEM_TOMATO_PICK_FROM_BUSH.get(), SoundSource.BLOCKS, 1.0F, 0.8F + level.random.nextFloat() * 0.4F);
            level.setBlock(pos, state.setValue(this.getAgeProperty(), 0), 2);
            return InteractionResult.SUCCESS;
        } else {
            return InteractionResult.FAIL;
        }
    }

    @Override
	public void playerDestroy(Level level, Player player, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity, ItemStack stack) {
		super.playerDestroy(level, player, pos, state, blockEntity, stack);
        int num = 1;
		if (this.isMaxAge(state)) {
			num += level.random.nextInt(1, 3);
		}
        popResource(level, pos, new ItemStack(ItemList.Chili.get(), num));
	}

}

File: src/main/kotlin/club/someoneice/thursday_delight/chili/ChiliCrop.kt
package club.someoneice.thursday_delight.chili

import club.someoneice.thursday_delight.RegisterInit
import net.minecraft.block.BlockState
import net.minecraft.block.Blocks
import net.minecraft.block.CropBlock
import net.minecraft.block.Material
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.ItemConvertible
import net.minecraft.item.ItemStack
import net.minecraft.sound.BlockSoundGroup
import net.minecraft.util.ActionResult
import net.minecraft.util.Hand
import net.minecraft.util.hit.BlockHitResult
import net.minecraft.util.math.BlockPos
import net.minecraft.world.BlockView
import net.minecraft.world.World
import java.util.*

class ChiliCrop() : CropBlock(Settings.of(Material.PLANT).strength(0.0f).sounds(BlockSoundGroup.CROP).noCollision()) {
    override fun onUse(state: BlockState, world: World, pos: BlockPos, player: PlayerEntity, hand: Hand, hit: BlockHitResult): ActionResult {
        val i = this.getAge(state)
        if (i >= maxAge) {
            world.setBlockState(pos, withAge(i - 2), 2)
            val itemDrop: Int = Random().nextInt(1, 3)
            player.inventory.offerOrDrop(ItemStack(RegisterInit.Chili, itemDrop))
        }

        return super.onUse(state, world, pos, player, hand, hit)
    }

    override fun canPlantOnTop(floor: BlockState, world: BlockView?, pos: BlockPos?): Boolean {
        return floor.isOf(Blocks.SOUL_SAND) || floor.isOf(Blocks.SOUL_SOIL)
    }

    override fun getSeedsItem(): ItemConvertible {
        return RegisterInit.Chili
    }
}